gateway/ln-gateway/src/lib.rs

```diff
@@ -1611,6 +1615,73 @@ impl Gateway {
         Ok(())
     }
 
+    /// Queries the client log for transaction events and returns to the user.
+    pub async fn handle_transactions_msg(
+        &self,
+        TransactionsPayload {
+            position,
+            limit,
+            federation_id,
+            event_kinds,
+        }: TransactionsPayload,
+    ) -> TransactionsResponse {
+        let federation_manager = self.federation_manager.read().await;
+        let configs = {
+            let mut dbtx = self.gateway_db.begin_transaction_nc().await;
+            dbtx.load_federation_configs().await
+        };
+
+        let clients = configs
+            .into_iter()
+            .filter_map(|(fed_id, _)| match federation_id {
+                Some(id) if id == fed_id => Some((
+                    id,
+                    federation_manager
+                        .client(&fed_id)
+                        .expect("Invalid federation")
+                        .value(),
+                )),
+                Some(_) => None,
+                None => Some((
+                    fed_id,
+                    federation_manager
+                        .client(&fed_id)
+                        .expect("Invalid federation")
+                        .value(),
+                )),
+            })
+            .collect::<Vec<_>>();
```

Comment:
Seems like this could be simplified by pulling out the match on federation id out of the iterator.